src/main/scala/edu/cmu/cs/obsidian/codegen/yulAST.scala

```diff
@@ -189,25 +223,58 @@ case class YulObject(name: String, code: Code, subobjects: Seq[YulObject], data:
         val freeMemPointer = 64 // 0x40: currently allocated memory size (aka. free memory pointer)
         val firstFreeMem = 128 //  0x80: first byte in memory not reserved for special usages
         // the free memory pointer points to 0x80 initially
-        var memoryInitRuntime: Expression = binary_ap("mstore", ilit(freeMemPointer), ilit(firstFreeMem))
+
+        var memoryInitRuntime: Expression = ap("mstore", ilit(freeMemPointer), ilit(firstFreeMem))
+
         var memoryInit: Expression = memoryInitRuntime
 
         def callValueCheck(): YulStatement = callvaluecheck
 
+        // together, these keep track of and create temporary variables for returns in the dispatch table
```

Comment:
Why do these need to be in a different namespace than other temps?

Reply:
I'm not sure, and actually I had what I thought was a bug where they weren't and it seemed to be fine, so maybe they don't. I kept them separate because it made it easier for me to debug what parts of the Yul we were outputting came from which logic, so it ends up being almost a readability thing.

That logic is repeated twice there, though, so there's a little bit of technical debt building up here.